### PR TITLE
Feat/badges slider

### PR DESCRIPTION
**Pull request by arathjz**
**What:**
- Create badges slider
- Add membership benefits to union's page

**Why:**
Relates to: https://app.asana.com/0/1159164196409129/1191556884196657/f

**How:**
- Create a slider for mobile and for desktop
- Integrate slider to membership benefits section
- Use react-pure-carousel for the slider

#### Media:
https://www.loom.com/share/834102914e8b4100a92031df17324035

### CODE REVIEW

File: src/components/BadgeSlider/DesktopSlider.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import { chunkArray } from 'utils/array';
+
+const ITEMS_PER_PAGE = 3;
+
+interface Props {
+  items: IBadge[];
+}
+
+const DesktopSlider: React.FC<Props> = ({ items }) => {
+  const [currentPage, setCurrentPage] = useState(1);
```
Comment (orlando) [inline src/components/BadgeSlider/DesktopSlider.tsx:16]: @arathjz while I think this should be easy enough, is there a library we can use for this?

File: tsconfig.json [hunk 2]
```diff
@@ -12,6 +12,7 @@
     "baseUrl": "./",
     "paths": {
       "components/*": ["src/components/*"],
```
Comment (orlando) [inline tsconfig.json:14]: maybe we want to use a symbol or something to indicate this path is an alias, something like `~`. No need to change this though

**Review by orlando — APPROVED** [review]
A couple of comments but LGTM. I love the fact we don't have a CSS file in this commit

File: src/components/BadgeSlider/DesktopSlider.tsx (hunk 1 above)
Comment (arathjz) [reply src/components/BadgeSlider/DesktopSlider.tsx:16]: I've heard about this one https://www.react-spring.io. We could give it a try

Comment (duranmla) [reply src/components/BadgeSlider/DesktopSlider.tsx:16]: react-spring is more for animations I think @orlando suggest in the slider itself something like https://github.com/brainhubeu/react-carousel so we add the dependency and then:

```
<Carousel plugins={['arrows']}>
    <img src={imageOne} />
    <img src={imageTwo} />
    <img src={imageThree} />
</Carousel>
```

_Maybe other with more starts but I guess the point is not having more code within the project for this._

File: src/utils/array.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const chunkArray = (arr: any, size: number) => {
```
Comment (duranmla) [inline src/utils/array.ts:2]: what's `chunkArray`? I have even seen the usage and is difficult to grasp what it does without reading implementation details. There is any better name we can assign? Or perhaps my English is failing?

File: static/icons/right-arrow.svg [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<svg width="26" height="64" viewBox="0 0 26 64" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (duranmla) [inline static/icons/right-arrow.svg:1]: I would say let's use material icons as much as we can. In that way, we will be able to keep some homogenous UI through all the apps without adding extra assets. Material icon `navigate_next` might work... Or it looks too bad?

**Review by duranmla — APPROVED** [review]
Couple minor things, check then out before merging. Thanks a lot!

**Comment by duranmla** [comment]
Also as seen on the video I think we should **avoid changing the viewport size** when moving from slide items, instead, let's fix the height to a maximum, and as a content producer, we should add content with a maximum of characters. This will limit a bit of the text we can add but will improve the UX.

File: src/utils/array.ts (hunk 3 above)
Comment (arathjz) [reply src/utils/array.ts:2]: Chunk is like splitting, I found that is actually a pretty common approach for this need. What this function allow us is to do the following:

```
const myArray = [1,2,3,4];

const chunkedArray = chunkArray(myArray, 2);
console.log(chunkedArray) // [ [1, 2], [3, 4] ]
```

File: tsconfig.json (hunk 2 above)
Comment (arathjz) [reply tsconfig.json:14]: Agree!

File: src/utils/array.ts (hunk 3 above)
Comment (orlando) [reply src/utils/array.ts:2]: What about using lodash for utilities like this, https://lodash.com/docs/#chunk, paired with https://github.com/lodash/lodash-webpack-plugin shouldn't affect the payload size too much

File: static/icons/right-arrow.svg (hunk 4 above)
Comment (arathjz) [reply static/icons/right-arrow.svg:1]: Looks a little bit different but could work

File: src/components/BadgeSlider/DesktopSlider.tsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import chunk from 'lodash.chunk';
```
Comment (orlando) [inline src/components/BadgeSlider/DesktopSlider.tsx:2]: I like this, we can keep like this until we want to add more functions

**Review by orlando — APPROVED** [review]
LGTM, good that we are using a lib for the slider and lodash 👍

**Pull request merged**